lib/tbot/config/configtemplate_ssh_host_cert.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"context"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/client"
+	"github.com/gravitational/teleport/lib/client/identityfile"
+	"github.com/gravitational/teleport/lib/tbot/bot"
+	"github.com/gravitational/teleport/lib/tbot/identity"
+	"github.com/gravitational/trace"
+)
+
+const (
+	// defaultSSHHostCertPrefix is the default filename prefix for the SSH host
+	// certificate
+	defaultSSHHostCertPrefix = "ssh_host"
+)
+
+// TemplateSSHClient contains parameters for the ssh_config config
+// template
+type TemplateSSHHostCert struct {
+	// Prefix is the filename prefix for the generated SSH host
+	// certificates
+	Prefix string `yaml:"prefix,omitempty"`
+
+	// Principals is a list of principals to request for the host cert.
+	Principals []string `yaml:"principals"`
+}
+
+func (c *TemplateSSHHostCert) CheckAndSetDefaults() error {
+	if c.Prefix == "" {
+		c.Prefix = defaultSSHHostCertPrefix
+	}
+
+	if len(c.Principals) == 0 {
+		return trace.BadParameter("at least one principal must be specified")
+	}
+
+	return nil
+}
+
+func (c *TemplateSSHHostCert) Name() string {
+	return TemplateSSHHostCertName
+}
+
+func (c *TemplateSSHHostCert) Describe(destination bot.Destination) []FileDescription {
+	ret := []FileDescription{
+		{
+			Name: c.Prefix,
+		},
+		{
+			Name: c.Prefix + "-cert.pub",
+		},
+	}
+
+	return ret
+}
+
+func (c *TemplateSSHHostCert) Render(ctx context.Context, bot Bot, currentIdentity *identity.Identity, destination *DestinationConfig) error {
+	dest, err := destination.GetDestination()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	// We'll need a client for the impersonated identity to request the certs.
+	botCfg := bot.Config()
+	authClient, err := bot.AuthenticatedUserClientFromIdentity(ctx, currentIdentity, botCfg.AuthServer)
```

Comment:
Does the bot.Config needs to be called to get the `botCfg.AuthServer` in order to inject it to the `AuthenticatedUserClientFromIdentity` ? 
Why not just reference to bot.config.AuthServer inside the `AuthenticatedUserClientFromIdentity` function ?

Reply:
Good point, we only ever pass it the configured value anyway. Unfortunately we still need to plumb it through to access `CertificateTTL`, though.